pydicom/pixel_data_handlers/jpeg_ls_handler.py

```diff
@@ -87,65 +90,34 @@ def get_pixeldata(dicom_dataset):
-                   dicom_dataset.file_meta.TransferSyntaxUID.name))
-        raise ImportError(msg)
-    # Make NumPy format code, e.g. "uint16", "int32" etc
-    # from two pieces of info:
-    # dicom_dataset.PixelRepresentation -- 0 for unsigned, 1 for signed;
-    # dicom_dataset.BitsAllocated -- 8, 16, or 32
-    if dicom_dataset.PixelRepresentation == 0:
-        format_str = 'uint{}'.format(dicom_dataset.BitsAllocated)
-    elif dicom_dataset.PixelRepresentation == 1:
-        format_str = 'int{}'.format(dicom_dataset.BitsAllocated)
-    else:
-        format_str = 'bad_pixel_representation'
-    try:
-        numpy_format = numpy.dtype(format_str)
-    except TypeError:
-        msg = ("Data type not understood by NumPy: "
-               "format='{}', PixelRepresentation={}, "
-               "BitsAllocated={}".format(
-                   format_str,
-                   dicom_dataset.PixelRepresentation,
-                   dicom_dataset.BitsAllocated))
-        raise TypeError(msg)
-
-    numpy_format = dtype_corrected_for_endianness(
-        dicom_dataset.is_little_endian, numpy_format)
-
-    # decompress here
-    UncompressedPixelData = bytearray()
-    if ('NumberOfFrames' in dicom_dataset and
-            dicom_dataset.NumberOfFrames > 1):
-        # multiple compressed frames
-        CompressedPixelDataSeq = pydicom.encaps.decode_data_sequence(
-            dicom_dataset.PixelData)
-        # print len(CompressedPixelDataSeq)
-        for frame in CompressedPixelDataSeq:
-            decompressed_image = jpeg_ls.decode(
-                numpy.frombuffer(frame, dtype=numpy.uint8))
-            UncompressedPixelData.extend(decompressed_image.tobytes())
+        raise ImportError(
+            "The jpeg_ls package is required to use pixel_array for this "
+            f"transfer syntax {tsyntax.name}, and jpeg_ls could not be "
+            "imported"
+        )
+
+    pixel_bytes = bytearray()
+
+    nr_frames = getattr(ds, "NumberOfFrames", 1) or 1
+    if nr_frames > 1:
+        for src in decode_data_sequence(ds.PixelData):
+            frame = jpeg_ls.decode(numpy.frombuffer(src, dtype='u1'))
+            pixel_bytes.extend(frame.tobytes())
     else:
-        # single compressed frame
-        CompressedPixelData = pydicom.encaps.defragment_data(
-            dicom_dataset.PixelData)
-        decompressed_image = jpeg_ls.decode(
-            numpy.frombuffer(CompressedPixelData, dtype=numpy.uint8))
-        UncompressedPixelData.extend(decompressed_image.tobytes())
-
-    pixel_array = numpy.frombuffer(UncompressedPixelData, numpy_format)
-    if should_change_PhotometricInterpretation_to_RGB(dicom_dataset):
-        dicom_dataset.PhotometricInterpretation = "RGB"
-
-    return pixel_array
+        src = defragment_data(ds.PixelData)
+        frame = jpeg_ls.decode(numpy.frombuffer(src, dtype='u1'))
+        pixel_bytes.extend(frame.tobytes())
+
+    arr = numpy.frombuffer(pixel_bytes, pixel_dtype(ds))
```

Comment:
This looks much nicer than before, but I don't really understand why the old code was as it was. I will just rely on the tests passing here. Not really related to typing...